docker/Dockerfile.alpine

```diff
@@ -0,0 +1,51 @@
+# -- build-environment --
+# see https://docs.docker.com/engine/userguide/eng-image/multistage-build/
+
+FROM alpine:3.7 AS build-env
+
+WORKDIR /build
+RUN     apk update && apk --no-cache upgrade && \
+        apk --no-cache add \
+            erlang \
+                erlang-asn1 erlang-compiler erlang-crypto erlang-dev \
+                erlang-diameter erlang-eunit erlang-hipe erlang-inets \
+                erlang-kernel erlang-parsetools erlang-public-key \
+                erlang-runtime-tools erlang-sasl erlang-ssl erlang-stdlib \
+                erlang-syntax-tools \
+            gcc \
+            git \
+            libc-dev libc-utils \
+            libgcc \
+            linux-headers \
+            make \
+            musl-dev musl-utils \
+            pcre2 \
+            pkgconf \
+            scanelf \
+            wget \
+            zlib
+
+RUN     wget -O /usr/local/bin/rebar3 https://s3.amazonaws.com/rebar3/rebar3 && \
+        chmod a+x /usr/local/bin/rebar3
+
+ADD     . /build
+RUN     rebar3 as prod,native get-deps && rebar3 as prod,native release
```

Comment:
get-deps is not a rebar3 task

Reply:
http://www.rebar3.org/docs/commands#section-get-deps

looks legit (and yes, "compile" or any other task which involves fetching the deps first would fetch the deps first. the separation allows to do $thing in between fetching and compiling, thats all.